azurecompute/src/main/java/org/jclouds/azurecompute/domain/DataVirtualHardDiskParam.java

```diff
@@ -0,0 +1,156 @@
+
+import java.net.URI;
+
+/**
+ * Create DataVirtualHardDisk
+ *
+ * @see <a href="http://msdn.microsoft.com/en-us/library/azure/jj157186.aspx#DataVirtualHardDisks" >api</a>
+ */
+
+@AutoValue public abstract class DataVirtualHardDiskParam {
+
+   /**
+    * Specifies the caching mode of the operating system disk.
+    * This setting impacts the consistency and performance of the disk.
+    * Possible values are:
+    * ReadOnly
+    * ReadWrite
+    * The default value is ReadWrite
+    */
+   @Nullable public abstract String hostCaching();
+
+   /**
+    * Specifies the description of the disk.
+    */
+   @Nullable public abstract String diskLabel();
+
+   /**
+    * Required if an existing disk is being used to create a Virtual Machine.
+    * Specifies the name of a new or existing disk
+    */
+   @Nullable public abstract String diskName();
+
+   /**
+    * Specifies the Logical Unit Number (LUN) for the data disk. If the disk is the first disk that is added,
+    * this element is optional and the default value of 0 is used. If more than one disk is being added,
+    * this element is required.
+    * <p/>
+    * You can use Get Role to find the LUN numbers that are already being used.
+    * Valid LUN values are 0 through 31
+    */
+   @Nullable public abstract Integer LUN();
+
+   /**
+    * Specifies the size, in GB, of an empty disk to be attached to the Virtual Machine.If the disk that is being added
+    * is already registered in the subscription, this element is ignored.If the disk and VHD is being created by Azure
+    * as it is added, this element defines the size of the new disk.
+    * <p/>
+    * The number of disks that can be added to a Virtual Machine is limited by the size of the machine.
+    * <p/>
+    * This element is used with the MediaLink element.
+    */
+   @Nullable public abstract Integer logicalDiskSizeInGB();
+
+   /**
+    * If the disk that is being added is already registered in the subscription or the VHD for the disk already exists
+    * in blob storage, this element is ignored. If a VHD file does not exist in blob storage, this element defines the
+    * location of the new VHD that is created when the new disk is added.
+    * Example:
+    * http://example.blob.core.windows.net/disks/mydatadisk.vhd
+    */
+   @Nullable public abstract URI mediaLink();
+
+   /**
+    * If the disk that is being added is already registered in the subscription or the VHD for the disk does not exist
+    * in blob storage, this element is ignored. If the VHD file exists in blob storage, this element defines the path to
+    * the VHD and a disk is registered from it and attached to the virtual machine.
+    */
+   @Nullable public abstract URI sourceMediaLink();
+
+   public Builder toBuilder() {
+      return builder().fromDataVirtualHardDiskParams(this);
+   }
+
+   public static Builder builder() {
+      return new Builder();
+   }
+
+   public static final class Builder {
```

Comment:
The google auto version we use now supports builders. We should try to move to using them.